packages/core/src/proxy_configuration.ts

```diff
@@ -264,21 +281,27 @@ export class ProxyConfiguration {
             password,
             hostname,
             port: port!,
+            proxyTier: tier,
         };
     }
 
     /**
      * Given a session identifier and a request / proxy tier, this function returns a new proxy URL based on the provided configuration options.
      * @param _sessionId Session identifier
      * @param options Options for the tiered proxy rotation
-     * @returns A string with a proxy URL.
+     * @returns An object with the proxy URL and the proxy tier used.
      */
-    protected _handleTieredUrl(_sessionId: string, options?: TieredProxyOptions): string {
+    protected _handleTieredUrl(_sessionId: string, options?: TieredProxyOptions): {
+        proxyUrl: string;
+        proxyTier?: number;
+    } {
```

Comment:
same here, this feels inconsistent, better to define the return type separately i guess